File: codeforcesBruta/q1/main.py
def calcular_media_provas(notas):
    soma = sum(notas)
    return soma / len(notas)

def obter_kesima_maior_nota(notas, k):
    notas.sort(reverse=True)
    return notas[k-1]

while True:
    try:
        n, k = map(int, input().split())
        notas = list(map(int, input().split()))

        media_provas = calcular_media_provas(notas)
        kesima_maior_nota = obter_kesima_maior_nota(notas, k)

        nota_final = (media_provas + kesima_maior_nota) / 2

        print(f"{nota_final:.1f}")
    except EOFError:
        break

File: codeforcesBruta/q1/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int N, K;

    while (cin >> N >> K)
    {
        vector<int> notas(N);

        for (int i = 0; i < N; i++)
        {
            cin >> notas[i];
        }

        sort(notas.begin(), notas.end(), greater<int>());

        double soma = 0;

        for (int i = 0; i < K; i++)
        {
            soma += notas[i];
        }

        double media = soma / K;

        cout << fixed;
        cout.precision(1);
        cout << media << endl;
    }

    return 0;
}
